src/pages/index.js

```diff
@@ -62,27 +108,135 @@ const editProfileModal = new PopupWithForm(
 );
 editProfileModal.setEventListeners();
 
+const editAvatarPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleAvatarEditSubmit
+);
+editAvatarPopup.setEventListeners();
+
+const deleteCardPopup = new PopupWithConfirmation("#confirm-delete-modal");
+deleteCardPopup.setEventListeners();
+
 function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", openImagePreview);
-  return cardElement.getView();
+  console.log("Creating card with data:", cardData);
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    openImagePreview,
+    handleLikeCard,
+    handleDeleteCard
+  );
+  const view = cardElement.getView();
+  console.log("Card element created:", {
+    id: cardElement.getCardId(),
+    element: view,
+  });
+  return view;
+}
+
+function handleDeleteCard(card) {
+  deleteCardPopup.setAction(() => {
+    api
+      .deleteCard(card.getCardId())
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error("Error deleting card:", err));
+  });
+  deleteCardPopup.open(card);
+}
+
+function handleLikeCard(card) {
+  if (card.isLiked) {
+    return api
+      .removeLike(card.getCardId())
+      .then(() => {
+        card.setCardLikes(false);
+      })
+      .catch((err) => console.error("Failed to dislike card:", err));
+  } else {
+    return api
+      .addLike(card.getCardId())
+      .then(() => {
+        card.setCardLikes(true);
+      })
+      .catch(console.error);
+  }
 }
 
 function handleProfileEditSubmit(formValues) {
-  userInfo.setUserInfo({
+  const apiData = {
     name: formValues.title,
     about: formValues.description,
-  });
-  editProfileModal.close();
+  };
+
+  editProfileModal.renderLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```